### DISCUSSION TITLE
RPI prime-util build: Non-Defined Cache Fields

### DISCUSSION THREAD

**Discussion by jake14000** [nasa/fprime#1702] [Q&A] [answered]
Hello,
I am running through the getting started tutorial and am working to get the Raspberry Pi system up and running. I have downloaded the RPI tools, moved them into /opt/rpi, and was able to fprime-generate within cd fprime/RPI

`(fprime-venv) myname@Names-MBP RPI % fprime-util generate`
`[ERROR] /Users/myname/fprime/RPI/build-fprime-automatic-raspberrypi already exists.`
However, once I run the fprime-build or gds, commands, I receive the following:

`(fprime-venv)myname@Names-MBP RPI % fprime-util build`
`[ERROR] /Users/myname/fprime/RPI/build-fprime-automatic-raspberrypi is an invalid F prime deployment. Does not define any of the cache fields: FPRIME_PROJECT_ROOT,FPRIME_LIBRARY_LOCATIONS,FPRIME_FRAMEWORK_PATH`
`(fprime-venv) myname@Names-MBP RPI % fprime-gds`
`[INFO] Autodetected artifacts root '/Users/myname/fprime/RPI/build-artifacts' from deployment settings.ini file.`
`[ERROR] binary location /Users/myname/fprime/RPI/build-artifacts/Darwin/bin does not exist`

Is there something I am missing here?

**Comment by LeStarch** [comment, marked as answer]
The RPI tools don't run on macOS.  Thus you are running `fprime-util generate raspberrypi` and it is failing because those tools only run on Linux.   Since the `fprime-util generate` step didn't work, you are receiving an error on `fprime-util build` and then `fprime-gds`.

In order to run the RPI tools you'll need to run on Linux via a VM, or container.

**Reply by jake14000** [reply]
Ahh, okay thank you. So how/where can I cross-compile F Prime onto some hardware for a CubeSat project? Where can I find some previous F Prime CubeSat deployments, as well as hardware specs for F Prime?

EDIT: I have read some F Prime literature that hints at CubeSat involvement (e.g., ASTERIA) but a lot of the details for their hardware specs or what they did to cross-compile onto that hardware I could not find.

**Reply by LeStarch** [reply]
Our sample hardware is the raspberry pi.  We have been working on a more complete reference example based on the raspberrypi.  It has been released as an initial version: [fprime-system-reference](https://github.com/fprime-community/fprime-system-reference).

Each cubesat that runs fprime has different processor specifications.  Some are really small, others are larger. How to run the cross-compilers for those cubesats differ depending on the project itself. Most projects that develop fprime cubesat deployments don't open source their full deployment, and as such we have been working on the above reference as a guide.
